TM1py/Services/CellService.py

```diff
@@ -621,43 +627,58 @@ def clear_from_df(self, cube: str, df: pd.DataFrame, dimension_mapping: Dict = N
 
         df = df.astype(str)
 
-        unique_entries = [{col_name: df[col_name].unique()} for col_name in df.columns]
+        elements_by_column = {col_name: df[col_name].unique() for col_name in df.columns}
```

Comment:
Hi @MariusWirtz,
the case and space insensitive is really an issue in general, but in this case that's why we are checking each column name if it matches with a dimension name from the cube [L637](https://github.com/cubewise-code/tm1py/blob/feature/clear_with_dataframe/TM1py/Services/CellService.py#L637). So it would raise an error and won't execute anything. Adding the check as you suggest, will help fail sooner or will be able to deal with ase and space insensitiveness